lib/puppet/rest/routes.rb

```diff
@@ -34,14 +34,28 @@ def self.get_certificate(client, name)
     # @param [String] name the crl to fetch
     # @raise [Puppet::Rest::ResponseError] if the response status is not OK
     # @return nil
-   def self.get_crls(client, name, &block)
-     ca.with_base_url(client.dns_resolver) do |base_url|
-       header = { 'Accept' => 'text/plain', 'Accept-Encoding' => ACCEPT_ENCODING }
-       client.get("#{base_url}certificate_revocation_list/#{name}", header: header) do |chunk|
-         block.call(chunk)
-       end
-     end
-   end
+    def self.get_crls(client, name, &block)
+      ca.with_base_url(client.dns_resolver) do |base_url|
+        header = { 'Accept' => 'text/plain', 'Accept-Encoding' => ACCEPT_ENCODING }
+        client.get("#{base_url}certificate_revocation_list/#{name}", header: header) do |chunk|
+          block.call(chunk)
+        end
+      end
+    end
 
+    # Make an HTTP request to send the named CSR, using the given
+    # HTTP client.
+    # @param [Puppet::Rest::Client] client the HTTP client to use to make the request
+    # @param [String] csr_pem the contents of the CSR to sent to the CA
+    # @param [String] name the name or the host whose CSR is being submitted
+    # @return nil
```

Comment:
Won't this return an HTTP::Message?

Reply:
Oh, yes. I'd forgotten how the non-streaming API worked.